comment_scan/replier.rb

```diff
@@ -0,0 +1,296 @@
+        return
+    end
+
+    regex = hg_comment[0]
+    types = hg_comment[1]
+    comments_to_display = []
+
+    if types == '*'
+        comments_to_display = Comment.where("fps >= ?", 1).select { |comment| %r{#{regex}}.match(comment.body_markdown.downcase) }
+    else
+        comments_to_display = Comment.where(post_type: types).where("fps >= ?", 1).select { |comment| %r{#{regex}}.match(comment.body_markdown.downcase) }
+    end
+
+    replier.chatter.say("There are no fp's on that howgood.", room_id) if comments_to_display.count == 0
+
+    #no need to return and prevent this from runnng. It won't do anything if db_comments is empty anyways
+    comments_to_display.take(num_to_display).each { |comment| replier.scanner.report_db_comment(comment, should_post_matches: false) }
+end
+
+def howgood_glob(replier, msg_id, parent_id, room_id, num_to_display=3)
+    hg_comment = MessageCollection::ALL_ROOMS.howgood_for(parent_id.to_i)
+    return if hg_comment.nil?
+
+    if !/\A\d+\z/.match(num_to_display) || num_to_display.to_i < 1
+        replier.chatter.say("Bad number. Reply to howgood with <comment_type> <num> to print num matches of comment_type where comment types are tp, fp, and *", room_id)
+        return
+    end
+
+    regex = hg_comment[0]
+    types = hg_comment[1]
+    comments_to_display = []
+
+    if types == '*'
+        comments_to_display = Comment.select { |comment| %r{#{regex}}.match(comment.body_markdown.downcase) }
+    else
+        comments_to_display = Comment.where(post_type: types).select { |comment| %r{#{regex}}.match(comment.body_markdown.downcase) }
+    end
+
+    replier.chatter.say("There are no comments on that howgood.", room_id) if comments_to_display.count == 0
+
+    #no need to return and prevent this from runnng. It won't do anything if db_comments is empty anyways
+    comments_to_display.take(num_to_display).each { |comment| replier.scanner.report_db_comment(comment, should_post_matches: false) }
+end
+
+#TODO: Add a "none" option for howgood at some point. Would work by checking that tps/fps = nil
+
+def bad_command(replier, msg_id, parent_id, room_id, *args)
+    db_comment = MessageCollection::ALL_ROOMS.comment_for(parent_id.to_i)
+    hg_comment = MessageCollection::ALL_ROOMS.howgood_for(parent_id.to_i)
+
+    if !db_comment.nil?
+        if args.length > 0 #They're not trying to give a command
+            #Maybe make conversation back (33% chance)
+            replier.chatter.say(":#{msg_id} #{replier.random_response}", room_id) if rand() > 0.67
+        else
+            replier.chatter.say("Invalid feedback type. Valid feedback types are tp, fp, rude, and wrongo", room_id)
+        end
+    elsif !hg_comment.nil?
+        replier.chatter.say("Invalid comment type. Reply to howgood with <comment_type> <num> to print num matches of comment_type where comment types are tp, fp, and *", room_id)
+    else
+        puts "That was not a report"
```

Comment:
I'd be careful about using `puts`. This means that messages get logged to the console, which really shouldn't be for anything except debugging info. I wouldn't print this.

Reply:
I've switched everything over to using Logger's at this point. This print is something that's been there since the beginning of time it seems. Do you want me to leave it as a debug statement? Or remove it entirely?

Reply:
Well, it existed as something the bot said, until we decided we didn't want it anymore, right? If it's not getting sent to the room, I see no use for it.

Reply:
Sounds good to me.